src/SFML/Graphics/Shader.cpp

```diff
@@ -156,11 +210,50 @@ Shader::CurrentTextureType Shader::CurrentTexture;
 
 
 ////////////////////////////////////////////////////////////
+struct Shader::UniformBinder : private sf::NonCopyable
+{
+    ////////////////////////////////////////////////////////////
+    /// \brief Constructor: set up state before uniform is set
+    ///
+    ////////////////////////////////////////////////////////////
+    UniformBinder(Shader* shader, const std::string& name) :
+    program(0),
+    location(-1)
+    {
+        if (shader->m_shaderProgram)
```

Comment:
Because ```GlResource``` is only suitable for actual resources? It does things in its constructor and destructor that are not wanted for such a local helper functor.